src/engine/components/sprite_renderer.rb

```diff
@@ -88,7 +88,15 @@ def set_shader_model_matrix
     end
 
     def set_shader_camera_matrix
-      camera_matrix = Engine::Camera.instance.matrix
+      # camera_matrix = Engine::Camera.instance.matrix
+      # Reverting this because something's wrong
+      # When the ship, or an asteroid, flies off the screen, it takes a while before it comes back onto the screen
+      camera_matrix = Matrix[
+         [2.0 / Engine.screen_width, 0, 0, 0],
+         [0, 2.0 / Engine.screen_height, 0, 0],
+         [0, 0, 1, 0],
+         [-1, -1, 0, 1]
+       ]
```

Comment:
This should be done by modifying the clamp to screen component which is probably using the old, hardcoded numbers, you can read them off the camera instead

Reply:
So I did a bit of digging, and noticed that it looks like the `OrthographicCamera` should be using framebuffer height and width, according to the demo `./simple.rb`:
``` ruby
# Window is set to 640 x 480
window = GLFW.CreateWindow(640, 480, "Simple example", nil, nil)

# Framebuffer returns 1280 x 960
GLFW.GetFramebufferSize(window, width_buf, height_buf)
width = width_buf.unpack1('L')
height = height_buf.unpack1('L')
```

Once I updated that (see [this commit](https://github.com/MaxHatfull/ruby_rpg/pull/8/commits/1c18da5840014d71a3bfbbb3fe876bd9528344ea)) it's now working nicely 👍

It might be worth renaming `update_screen_size`,`screen_width` and `screen_height` to include `framebuffer`.
Happy to do this as part of #11, or as a follow up if you're keen?